### DISCUSSION TITLE
How to get VertexAi citations

### DISCUSSION THREAD

**Discussion by Goldritter** [langchain4j/langchain4j#1583] [Q&A]
Hi, 

I try to use the VertexAiGeminiChatModel to answer a question and verify the answer with an attached Datastore. 

It does work, but the document citations are missing in the AiMessage. The received Response  contains the citations in grounding_metadata part but the generate Method does not extract these information. 

Does anybody has an idea how I can get these Info with the Options from Langchain?

Best regards Marcus

**Comment by AntonLGVS** [comment]
I would like to join this question because I also need this functionality but for AzureOpenAIModel. Currently, we are not able to get this using plain Java without AOP or proxing search client.
If you have a look at image, you will see that there no any place to inject custom handler or listener to serve citations.
![image](https://github.com/user-attachments/assets/5690d04a-ab52-4657-bc6e-88720b27ba0d)
